llama_index/vector_stores/elasticsearch.py

```diff
@@ -0,0 +1,449 @@
+            List of node IDs that were added to the index.
+
+        Raises:
+            ImportError: If elasticsearch python package is not installed.
+            BulkIndexError: If Elasticsearch bulk indexing fails.
+        """
+        try:
+            from elasticsearch.helpers import BulkIndexError, bulk
+        except ImportError:
+            raise ImportError(
+                "Could not import elasticsearch python package. "
+                "Please install it with `pip install elasticsearch`."
+            )
+
+        if len(embedding_results) == 0:
+            return []
+
+        if create_index_if_not_exists:
+            dims_length = len(embedding_results[0].embedding)
+            self._create_index_if_not_exists(
+                index_name=self.index_name, dims_length=dims_length
+            )
+
+        embeddings: List[List[float]] = []
+        texts: List[str] = []
+        metadatas: List[dict] = []
+        ids: List[str] = []
+        for node in embedding_results:
+            ids.append(node.id)
+            embeddings.append(node.embedding)
+            texts.append(node.node.get_content(metadata_mode=MetadataMode.NONE))
+            metadatas.append(node_to_metadata_dict(node.node, remove_text=True))
+
+        requests = []
+        return_ids = []
+
+        for i, text in enumerate(texts):
+            metadata = metadatas[i] if metadatas else {}
+            _id = ids[i] if ids else str(uuid.uuid4())
+            request = {
+                "_op_type": "index",
+                "_index": self.index_name,
+                self.vector_field: embeddings[i],
+                self.text_field: text,
+                "metadata": metadata,
+                "_id": _id,
+            }
+            requests.append(request)
+            return_ids.append(_id)
+
+        bulk(self.client, requests, chunk_size=self.batch_size, refresh=True)
+        try:
+            success, failed = bulk(self.client, requests, stats_only=True, refresh=True)
+            logger.debug(f"Added {success} and failed to add {failed} texts to index")
+
+            logger.debug(f"added texts {ids} to index")
+            return return_ids
+        except BulkIndexError as e:
+            logger.error(f"Error adding texts: {e}")
+            firstError = e.errors[0].get("index", {}).get("error", {})
+            logger.error(f"First error reason: {firstError.get('reason')}")
+            raise e
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """Delete node from Elasticsearch index.
+
+        Args:
+            ref_doc_id: ID of the node to delete.
+            delete_kwargs: Optional. Additional arguments to pass to Elasticsearch delete_by_query.
+
+        Raises:
+            Exception: If Elasticsearch delete_by_query fails.
+        """
+
+        try:
+            self.client.delete_by_query(
+                index=self.index_name,
+                query={"match": {"_id": ref_doc_id}},
```

Comment:
actually, you'll want to match by `doc_id` in the metadata to delete. This deletes all nodes associated with a single input document (i.e. the ref doc)

Is this possible to do here?

Reply:
thanks! Updated PR with tests on how to do this :).

Reply:
Thanks!